dsl/camel-kamelet-main/src/main/java/org/apache/camel/main/KameletMain.java

```diff
@@ -732,6 +734,21 @@ protected CamelContext createCamelContext() {
             throw RuntimeCamelException.wrapRuntimeException(e);
         }
 
+        // Start the context service loader plugin after all dependencies and downloaders are set up
+        // This ensures the classpath is enhanced and ServiceLoader can discover third-party plugins
```

Comment:
ok so this force these to be loader earlier than usual (than standard in camel-core) and this allows camel-jbang etc to have custom plugins, for example also download dependencies and whatnot during their startup. Otherwise this kind of code is executed in camel-core already.

????